movie-app/src/shared-components/movie-item/MovieItem.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useCallback } from "react";
+import MovieItemWrapper from "./MovieItemWrapper";
+import posterLogo from "../../assets/images/netflix-streaming-vs-traditional-cable.jpg";
+import { IMovie } from "../../interfaces/Movie";
+
+const MovieItem = (props: IMovie) => {
+  const navigate = useCallback(() => {
+    console.log(`navigated to details with + ${props.id}`);
+  }, []);
+
+  return (
+    <MovieItemWrapper onClick={() => navigate()}>
```

Comment:
`useCallback` doesn't really helpse here: you're creating a new function on each render. Passing `navigate` as is would be enough